app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -17,6 +19,23 @@ module TwoFactorAuthenticatable
 
   private
 
+  def apply_secure_headers_override
+    return unless after_sign_in_path_for.start_with?(openid_connect_authorize_path)
```

Comment:
I guess this guard clause keeps out the non-Openid requests, so 🤷‍♀️  ok seems like the least evil thing to do.